Lib/test/test_email/test_email.py

```diff
@@ -3342,9 +3342,10 @@ def test_make_msgid_idstring(self):
             '.test-idstring@testdomain-string>')
 
     def test_make_msgid_default_domain(self):
```

Comment:
Unfortunately this change removes the whole point of the test, which is to check what happens when the domain parameter *isn't* being provided.  I think instead the test should just mock socket.getfqdn(), something like this (untested by me):
```
import mock   # at top of module

   with mock.patch('socket.getfqdn') as mock_getfqdn:
       mock_getfqdn.return_value = domain = 'pythontest.example.com'
       self.assertTrue( ... ) 
```

But I do think this is worth fixing.

Reply:
Sorry for my misunderstanding over tests concept. Will apply your suggestion, thanks.

Reply:
The new version looks good to me; thanks!